sources/api/shibaken/src/main.rs

```diff
@@ -43,10 +43,14 @@ impl UserData {
 async fn fetch_public_keys_from_imds() -> Result<Vec<String>> {
     info!("Connecting to IMDS");
     let mut client = ImdsClient::new().await.context(error::ImdsClient)?;
-    client
+    match client
         .fetch_public_ssh_keys()
         .await
-        .context(error::ImdsClient)
+        .context(error::ImdsClient)?
+    {
+        Some(public_keys) => Ok(public_keys),
+        None => Ok(Vec::new()),
+    }
```

Comment:
This should work and do the same thing:

```suggestion
        .context(error::ImdsClient)?
        .unwrap_or_else(|| Vec::new());
```

Reply:
👍, but clippy preferred `.unwrap_or_else(Vec::new)`